Program.cs

```diff
@@ -9,72 +9,85 @@ class Program
+                Console.WriteLine($"Example: {args[0]} 1000 s3 /path1/on/s3 s3 /path2/on/s3");
+                Console.WriteLine($"Example: {args[0]} 1000 s3 /path/on/s3 gpkg geo.gpkg");
                 return;
             }
 
-            string baseType = args[1];
-            string newType = args[3];
-            string basePath = args[2];
-            string newPath = args[4];
-            int batchSize = int.Parse(args[5]);
-
-            // char* mergeType = getMergeType();
-            // if (mergeType == "GPKG")
-            // {
-            //     File.Exists();
-            //     File.Exists();
-            //     startGpkgProccess(basePath, newPath, batchSize);
-            // }
-            // else
-            // {
-            //     Directory.Exists();
-            //     Directory.Exists();
-            //     startFolderProccess(basePath, newPath, batchSize);
-            // }
-
-            // Data baseData = new Gpkg(fullBasePath, 1000);
-            // Data newData = new Gpkg(fullNewPath, 1000);
-
-            Data baseData = Data.CreateDatasource(baseType, basePath, batchSize);
-            Data newData = Data.CreateDatasource(newType, newPath, batchSize);
-
-            if (!baseData.Exists())
-            {
-                Console.WriteLine("Base data does not exist.");
-                return;
-            }
+            int batchSize = int.Parse(args[1]);
+            string baseType = args[2];
+            string basePath = args[3];
 
-            if (!newData.Exists())
-            {
-                Console.WriteLine("New data does not exist");
-                return;
-            }
+            int sourceIndex = 4;
 
-            bool validate = bool.Parse(Configuration.Instance.GetConfiguration("GENERAL", "validate"));
-            if (validate)
-            {
-                Src.Proccess.Validate(baseData, newData);
-            }
-            else
+            while (args.Length > sourceIndex + 1)
             {
+                Stopwatch stopWatch = new Stopwatch();
+                stopWatch.Start();
+
+                string newType = args[sourceIndex];
+                string newPath = args[sourceIndex + 1];
+
+                // Console.WriteLine($"type: {newType}, path: {newPath}");
+                // sourceIndex += 2;
+                // continue;
+
+                Data baseData = Data.CreateDatasource(baseType, basePath, batchSize);
+                Data newData = Data.CreateDatasource(newType, newPath, batchSize);
+
+                if (!baseData.Exists())
+                {
+                    Console.WriteLine("Base data does not exist.");
+                    return;
+                }
+
+                if (!newData.Exists())
+                {
+                    Console.WriteLine("New data does not exist");
+                    return;
+                }
+
                 Src.Proccess.Start(baseData, newData);

```

Comment:
the logic itself should support multi sources instead of calling it multiple times.

merging all tile of the same coordinates at once can result in vastly better performance if the merge is optimized (if u start merging from top image to the button you can skip lower images pixels for pixels that exists in upper level, more complex optimization can be to use lazy loading for the tiles and not having to read lower tiles at all if they are completely hidden this can result in huge improvments- for future upgrades)